flake.nix

```diff
@@ -0,0 +1,58 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.05";
+    crane.url = "github:ipetkov/crane";
+    crane.inputs.nixpkgs.follows = "nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, crane, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs {
+          inherit system;
+        };
+
+        craneLib = crane.lib.${system};
+
+        commonArgs = {
+          src = ./.;
+
+          buildInputs = with pkgs; [
+            openssl
+            pkg-config
+            perl
+          ];
+
+          nativeBuildInputs = with pkgs; [
+            pkg-config
+          ];
+        };
+
+        cargoArtifacts = craneLib.buildDepsOnly (commonArgs // {
+          pname = "minimint";
+        });
+
+        myCrateClippy = craneLib.cargoClippy (commonArgs // {
```

Comment:
I think `myCrateClippy` is just an artifact from some example, right? We can probably give it a better name.

Reply:
> I think `myCrateClippy` is just an artifact from some example, right? We can probably give it a better name.

I kept it the default name to avoid confusion we can change it tho.